setup.sh

```diff
@@ -1,6 +1,30 @@
 #!/usr/bin/env bash
 #encoding=utf8
 
+# This should be deleted as soon as a gym release with a fixed setuptools is made
+function check_is_new_setuptools_installed() {
+  stoolversion=$(${PYTHON} -m pip show setuptools | grep "Version:")
+  stoolversion=${stoolversion##* }
+  stoolversion=$(echo $stoolversion | grep -Eo "^[0-9]+")
+
+  if [ $stoolversion -gt 65 ]; then
+    echo "You are using a new version of setuptools. This is not supported by gym at the moment."
+    read -p "Do you want to downgrade setuptools to a version that is supported by gym? [y/N] "
+
+    if [[ $REPLY =~ ^[Yy]$ ]]
+    then
+      ${PYTHON} -m pip install setuptools==65.5.0
```

Comment:
If i'm honest, then my approach would've been to force this downgrade if freqai-rl (or dev) is selected (we do something similar with cython for raspberry installations).

It's not really optional - and we know for a fact that installation will fail if this requirement is not matched - so we're basically asking the user "would you like to have the installation fail - or not?" - without saying it out loud 😆

Reply:
So that's a funny thing. In the fresh, newly created venv the version is actually `59.6.0` it doesn't even pass the check if that check is after `source .env/bin/activate`. Maybe we can only leave it in the freqai requirements for the setup.sh. It will still not work if trying to install system-wide (through `pip install -r ....`). It will only work in setup.sh.

Reply:
Still again even if the version is `59.6.0` in venv if the version is not fixed in the requirement it doesn't compile gym.

Reply:
Ok, can't get simpler than that, just let people know in the docs haha

Reply:
that obviously works 😆 

i did a few tests (always on clear systems) - and found a working method for the setup.sh script.
combined with the documentation - that will have to suffice.

Reply:
Looks great, thank you for getting to the bottom of this :)